### DISCUSSION TITLE
How to add coupons to stripe's checkout

### DISCUSSION THREAD

**Discussion by Temp-insta** [community/community#36423] [Apps, API and Webhooks] [answered]
how to pass coupons to stripe.redirectToCheckout() method, don't see anything related to this in the documentation.

code sample
```javascript
    stripe.redirectToCheckout({
      items: plans_and_addons,
      successUrl: successUrl,
      cancelUrl: cancelUrl,
      clientReferenceId: customer_id, // helpful in webhook response
    })
```

**Comment by santosharron** [comment, marked as answer]
It's quite easy. You find the official document [here](https://stripe.com/docs/payments/checkout/discounts). Now let's solve your issue
You need to add `allow_promotional_code` value before your `successurl`. After changing that your code should look something like this.
```javascript
      stripe.redirectToCheckout({
      items: plans_and_addons,
      allow_promotion_codes: true,
      successUrl: successUrl,
      cancelUrl: cancelUrl,
      clientReferenceId: customer_id, // helpful in webhook response
    })
```

**Reply by Temp-insta** [reply]
Thanks a lot.
